src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryBase.java

```diff
@@ -248,16 +279,60 @@ public void verifyTheBigQueryValidationErrorMessageForInvalidProperty(String pro
   }
 
+  @Then("Verify the BigQuery validation error message for reference name {string}")
+  public void VerifytheBigQueryvalidationerrormessageforreferencename (String property) {
+    CdfStudioActions.clickValidateButton();
+    String expectedErrorMessage;
+    if (property.equalsIgnoreCase("gcsChunkSize")) {
+      expectedErrorMessage = PluginPropertyUtils
+              .errorProp(E2ETestConstants.ERROR_MSG_BQ_INCORRECT_CHUNKSIZE);
+    } else if (property.equalsIgnoreCase("bucket")) {
+      expectedErrorMessage = PluginPropertyUtils
+              .errorProp(E2ETestConstants.ERROR_MSG_BQ_INCORRECT_TEMPORARY_BUCKET);
+    } else if (property.equalsIgnoreCase("table")) {
+      expectedErrorMessage = PluginPropertyUtils
+              .errorProp(E2ETestConstants.ERROR_MSG_INCORRECT_TABLE_NAME);
+    }
+    else if(property.equalsIgnoreCase("referenceName")) {
+      expectedErrorMessage = PluginPropertyUtils
+              .errorProp(E2ETestConstants.ERROR_MSG_INCORRECT_REFERENCENAME);
+    }
+    else {
+      expectedErrorMessage = PluginPropertyUtils.errorProp(E2ETestConstants.ERROR_MSG_INCORRECT_REFERENCENAME).
+              replaceAll("referenceName", property.substring(0, 1).toUpperCase() + property.substring(1));
+    }
+    String actualErrorMessage = PluginPropertyUtils.findPropertyErrorElement("referenceName").getText();
+    System.out.println(actualErrorMessage);
+    Assert.assertEquals(expectedErrorMessage, actualErrorMessage);
+    String actualColor = PluginPropertyUtils.getErrorColor(PluginPropertyUtils.findPropertyErrorElement("referenceName"));
+    String expectedColor = ConstantsUtil.ERROR_MSG_COLOR;
+    Assert.assertEquals(expectedColor, actualColor);
+  }
+
+
+
   @Then("Validate records transferred to target table is equal to number of records from source table")
   public void validateRecordsTransferredToTargetTableIsEqualToNumberOfRecordsFromSourceTable()
-    throws IOException, InterruptedException {
+          throws IOException, InterruptedException {
     int countRecordsTarget = BigQueryClient.countBqQuery(TestSetupHooks.bqTargetTable);
     Optional<String> result = BigQueryClient.getSoleQueryResult("SELECT count(*)  FROM `" +
-                                                                  (PluginPropertyUtils.pluginProp("projectId"))
-                                                                  + "." + (PluginPropertyUtils.pluginProp
-      ("dataset")) + "." + TestSetupHooks.bqTargetTable + "` ");
+            (PluginPropertyUtils.pluginProp("projectId"))
+            + "." + (PluginPropertyUtils.pluginProp
+            ("dataset")) + "." + TestSetupHooks.bqTargetTable + "` ");
     int count = result.map(Integer::parseInt).orElse(0);
     BeforeActions.scenario.write("Number of records transferred from source table to target table:" + count);
     Assert.assertEquals(count, countRecordsTarget);
   }
+
+  @Then("Enter BigQuery property output schema {string} as macro argument {string}")
```

Comment:
This is repeated step, and also as mentioned remove it as we already have an existing step to select macro action for output schema, use that. Always try to use the existing steps if we already have. Do not create the step definition again.